### PR TITLE
CATROID-1098 REFACTOR ProjectRenameTask to Kotlin

### PR DESCRIPTION
**Pull request by petbuer**
https://jira.catrob.at/browse/CATROID-1098

REFACTOR ProjectRenameTask to Kotlin

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by qlin93 — APPROVED** [review]

File: catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectRenamer.kt [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    ProjectManager.getInstance().moveChangedFlag(projectDirectory.name, destinationName)
+    return destinationDirectory
+}
+
+fun tryRenameTo(projectDirectory: File, destinationDirectory: File): Boolean {
+    return try {
+        projectDirectory.renameTo(destinationDirectory)
+        true
+    } catch (e: IOException) {
+        Log.e(
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectRenamer.kt:86]: pls change to 
```suggestion
        Log.e(TAG, "Cannot rename project directory ${projectDirectory.absolutePath} to$destinationName", e)
```

File: catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectRenamer.kt [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    }
+}
+
+fun tryRenameProject(projectDirectory: File, destinationName: String): Boolean {
+    return try {
+        XstreamSerializer.renameProject(projectDirectory, destinationName)
+        true
+    } catch (e: IOException) {
+        Log.e(
+            TAG,
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectRenamer.kt:102]: same

File: catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectRenamer.kt [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        false
+    }
+}
+
+fun tryRenameProject(projectDirectory: File, destinationName: String): Boolean {
+    return try {
+        XstreamSerializer.renameProject(projectDirectory, destinationName)
+        true
+    } catch (e: IOException) {
+        Log.e(
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectRenamer.kt:101]:
```suggestion
       Log.e(TAG, "Cannot rename project directory ${projectDirectory.absolutePath} to$destinationName", e)
```

File: catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectRenamer.kt (hunk 1 above)
Comment (petbuer) [reply catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectRenamer.kt]: 👍

File: catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectRenamer.kt (hunk 3 above)
Comment (petbuer) [reply catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectRenamer.kt]: 👍

**Comment by ndrnour** [comment]
Thanks!

**Pull request merged**